iree/test/e2e/regression/lowering_config.mlir

```diff
@@ -1,9 +1,9 @@
 #compilation0 = #iree_codegen.compilation_info<
-    lowering_config = <tile_sizes = [[32, 32], [8, 8, 0], [0, 0, 8]]>,
+    lowering_config = <tile_sizes = [[32, 32, 0], [8, 8, 0], [0, 0, 8]]>,
```

Comment:
If we relax the verifier, this change is unnecessary.

Reply:
This is needed for some reasons. It's really tricky. Say that we have fill + matmul ops.

If only two tiling sizes are set, the strategy would use `fill` op as the anchor op. This stops fusion between fill + matmul.

If the three tiling sizes are set, the `matmul` op would be the anchor op. Then the fusion works.

Reply:
That is strange, I'd rather fix that.... Couple of questions that can help diagnose
1) Which operation has the attribute? Do both fill and matmul have the attribute? If thats the case, thats the bug (potentially here https://github.com/google/iree/blob/d487b5854bf38c8b9466899c583a081ad3d59bd8/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp#L879)
2) If not the bug is probably here (https://github.com/google/iree/blob/d487b5854bf38c8b9466899c583a081ad3d59bd8/iree/compiler/Codegen/Sandbox/LinalgTensorCodegenDriver.cpp#L54).

Could we look into that before we go ahead with this?

Reply:
1) Only matmul op carries the attribute -- which is expected.

One of the issues is how root op is determined. I checked that it is bottom up approach when anchor-op is not specified. In this case, linalg.matmul is picked in the first place. However the pattern is not kicked in. (I also verified it with manually specifying the anchor op to be linalg.matmul)

https://github.com/llvm/llvm-project/blob/5c116d50e42f93ef4fa9a9719121378ec6b99b69/mlir/lib/Dialect/Linalg/Transforms/LinalgStrategyPasses.cpp#L68-L72

The issue turns into why tile and fuse does not work when the anchor-op is linalg.matmul. This is the debug message:

```
Processing operation : 'linalg.matmul'(0xccdfff0) {
  * Pattern mlir::linalg::LinalgTileAndFuseTensorOpsPattern : 'linalg.matmul -> ()' {
Trying to match "mlir::linalg::LinalgTileAndFuseTensorOpsPattern"
    ** Failure : expect #tile sizes >= #loops
"mlir::linalg::LinalgTileAndFuseTensorOpsPattern" result 0
  } -> failure : pattern failed to match
} -> failure : pattern failed to match
```

There is a check that the tiling sizes should be equal to or greater than the number of loops. I think it is a relaxable condition because it is acceptable during tiling. We append zeros when the number of tiling sizes is less than the number of loops. I'm not sure why we have this condition upstream. @gyist should weigh in since he has more context about fusion.

Reply:
Here is the check: https://github.com/llvm/llvm-project/blob/5c116d50e42f93ef4fa9a9719121378ec6b99b69/mlir/lib/Dialect/Linalg/Transforms/Transforms.cpp#L561-L571

Reply:
Actually that might be a bug on our side. We should pick up the configuration from matmul, but the first linalg op in the bottum up tranversal should be the root.